### PR TITLE
A generator for Buddy FW headers and some MMU fixes

### PR DESCRIPTION
**Pull request by lukash**
BFW-3046

### CODE REVIEW

**Review by Derevin — CHANGES_REQUESTED** [review]

File: generate_buddy_headers.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+
+            inline constexpr uint8_t ERR_PRINTER_CODE = {printer_code};
+
+            typedef enum : uint16_t {{
+                ERR_UNDEF = 0,
+                {enum_items}
+            }} ErrorNumber;
+
+            typedef struct {{
```
Comment (Derevin) [inline generate_buddy_headers.py:131]: Is there a reason why ErrorDesc and ErrorNumber aren't 'normal' c++ structs? Usage of `constexpr` implies c++ therefore i'd expect it to be the same(-ish) as with mmu

Comment (lukash) [reply generate_buddy_headers.py:131]: I assumed the header is used in C somewhere and haven't realized C doesn't have `constexpr`. I haven't dug too deep in this regard and mostly kept what's in `errors_list.h` now. I can change it though (and use the `.hpp` suffix for the header).

File: generate_buddy_headers.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            #include "inttypes.h"
+            #include "i18n.h"
+            #include "button_operations.h"
+
+
+            namespace MMU2 {{
+
+            inline constexpr uint8_t ERR_MMU_CODE = {printer_code};
+
+            enum ErrorNumber : uint16_t {{
```
[the remark this answers is not in the document]
Comment (lukash) [reply generate_buddy_headers.py:98]: Yes, this is a bit bigger change though. `enum class` is strongly typed, so explicit conversions will be needed. You're right though it'd quite probably be better.

**Comment by lukash** [comment]
I'll change the struct definitions to C++ and try to switch to `enum class`es.

File: generate_buddy_headers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Derevin) [reply generate_buddy_headers.py:98]: Yeah, while i'm not too completely happy with how `enum class`es make some operations verbose, from experience it was always worth it over 'passing int to enum parameters' errors that are way more insidious

**Comment by lukash** [comment]
I/ve switched to C++ struct definitions, `enum class` and unified the naming of types. Should be GTG now.

**Review by DRracer — APPROVED** [review]

**Pull request merged**